java/core/src/main/java/com/microsoft/bond/Throw.java

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import java.util.Locale;
+
+/**
+ * Encapsulates Bond exception throwing and error messages.
+ * Some message details (e.g. list indices, map keys) may be added to
+ * exception message by certain specialized implementations of Bond type.
+ */
+final class Throw {
+
+    // invariant locale
+    private static final Locale LOCALE = Locale.ROOT;
+
+    // general entry point to raise a data exception on serialization or deserialization
+    static void raiseInvalidDataError(
+            Exception cause,
+            String format,
+            Object... args) throws InvalidBondDataException {
+        String message = String.format(LOCALE, format, args);
```

Comment:
To be consistent. These strings may reach end user if there is an exception. It's a good practice to keep invariant (root) locale for exception text.

---
In reply to: [123379360](https://github.com/Microsoft/bond/pull/502#discussion_r123379360) [](ancestors = 123379360)